mathics/builtin/numbers/calculus.py

```diff
@@ -812,6 +812,193 @@ def to_sympy(self, expr, **kwargs):
         except Exception:
             return None
 
+ class Reduce(Builtin):
+    
+    """
+    <dl>
+       <dt>'Reduce[$eqs$, $vars$]'
+       <dd> Reduces the statements in $eqs$ by solving equations in $vars$ and eliminating quantifiers.
+    </dl>
+    >> Reduce[x y == 0,{x, y}]
+     = x == 0  ||   y == 0 
+    >> Reduce[{x y == 0, y == 4},{x, y}]
+     = y == 4 && x == 0
+    >> Reduce[{x y == 0, y > 4},{x, y}]
+     =  x == 0 && y > 4
+    >> Reduce[x*y>0,{x,y}]
+     = (x < 0 && y < 0) || (x > 0 && y > 0)
+    >> Reduce[{x*y > 0, y>0},{x,y}]
+     =  x > 0 && y > 0
+    
+    >> Reduce[a<b,a]
+     =  b ∈ Reals && a < b
+    """
+    
+    messages = {
+        "eqf": "`1` is not a well-formed equation.",
+        "svars": 'Equations may not give solutions for all "solve" variables.',
+    }
+
+    rules = {
+        "Reduce[eqs_, vars_, Complexes]": "Reduce[eqs, vars]",
+       
+    }
+
+    summary_text = "find generic solutions for variables, in logical expression form"
+    
+    
+    def apply(self, eqs, vars, evaluation):
+        "Reduce[eqs_, vars_]"
+        
+        vars_original = vars
+        head_name = vars.get_head_name()
+        if head_name == "System`List":
+            vars = vars.leaves
+        else:
+            vars = [vars]
+        for var in vars:
+            if (
+                (var.is_atom() and not var.is_symbol())
+                or head_name in ("System`Plus", "System`Times", "System`Power")  # noqa
+                or constant & var.get_attributes(evaluation.definitions)
+            ):
+
+                evaluation.message("Solve", "ivar", vars_original)
+                return
+        eqs_original = eqs
+        if eqs.get_head_name() in ("System`List", "System`And"):
+            eqs = eqs.leaves
+        else:
+            eqs = [eqs]
+        sympy_eqs = []
+        sympy_denoms = []
+        for eq in eqs:
```

Comment:
Notice that ``eq`` is a Mathics  ``Expression``, but in line 882 it becomes into a sympy ``Expression``.

Reply:
So? This is exactly what is supposed to happen.

Reply:
Yes, but for clarity, it would be useful the change of name.

Reply:
For me, the more variables you add, the more confusing it gets. This comes from Solve, though.

Reply:
In the first ``if``, there was a ``pass`` instead a ``continue`` instruction. That produced that a (mathics) expression get into the list of simpy expressions, which at the end raised a weir error. If we use another name for the variable, it would be simpler to track that what really happened was that the variable was never set.